tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/ContextPropagation.cs

```diff
@@ -84,5 +89,62 @@ public void Set(StringBuilder carrier, string key, string value)
                 carrier.AppendFormat("\"{0}\":\"{1}\",", key, value);
             }
         }
+
+        /// <summary>
+        /// The adapter to use to append stuff to a string builder where a json is being built
+        /// </summary>
+        private readonly struct StringBuilderJsonAdapter : IBinaryHeadersCollection
+        {
+            private readonly StringBuilder _carrier;
+
+            public StringBuilderJsonAdapter(StringBuilder carrier)
+            {
+                _carrier = carrier;
+            }
+
+            public byte[] TryGetLastBytes(string name)
+            {
+                throw new NotImplementedException("this adapter can only be use to write to a StringBuilder, not to read data");
+            }
+
+            public void Add(string key, byte[] value)
+            {
+                _carrier.AppendFormat("\"{0}\":\"{1}\",", key, Convert.ToBase64String(value));
```

Comment:
nit: the format requires a little extra work in general to format the string etc, so it's _probably_ better to make the extra calls manually

```suggestion
                _carrier
                    .Append('"')
                    .Append(key)
                    .Append("\":\"")
                    .Append(Convert.ToBase64String(value))
                    .Append("\",");
```